src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -480,12 +485,90 @@ private static String createGCSBucketWithFilesAndFolder(String folderPath) throw
     return bucketName;
   }
 
+  @Before(order = 1, value = "@PUBSUB_SOURCE_TEST")
+  public static void createSourcePubSubTopic() throws IOException {
+    pubSubSourceTopic = "cdf-e2e-test-" + UUID.randomUUID();
+    PubSubClient.createTopic(pubSubSourceTopic);
+    BeforeActions.scenario.write("Source PubSub topic " + pubSubSourceTopic);
+  }
+
+  @Before(order = 1, value = "@PUBSUB_SCHEMA_TEST")
+  public static void createSourcePubSubSchema() throws IOException {
+    pubSubSchemaId = "cdf-e2e-test-" + UUID.randomUUID();
+    PubSubClient.createAvroSchema(pubSubSchemaId, PluginPropertyUtils.pluginProp("avrofile"));
+    BeforeActions.scenario.write("Source Schema " + pubSubSchemaId);
+  }
+
+  @Before(order = 2, value = "@PUBSUB_SCHEMA_TOPIC_TEST")
+  public static void createSourcePubSubSchemaTopic() throws IOException, InterruptedException {
+    pubSubSourceTopic = "cdf-e2e-test-" + UUID.randomUUID();
+    PubSubClient.createTopicWithSchema(pubSubSourceTopic, pubSubSchemaId, Encoding.BINARY);
+    BeforeActions.scenario.write("Schema Topic " + pubSubSourceTopic);
+  }
+
+  @Before(order = 3, value = "@PUBSUB_SUBSCRIPTION_TEST")
+  public static void createSubscriptionPubSubTopic() throws IOException {
+    pubSubSourceSubscription = "cdf-e2e-test-" + UUID.randomUUID();
+    PubSubClient.createSubscription(pubSubSourceSubscription, pubSubSourceTopic);
+    BeforeActions.scenario.write("Source PubSub subscription " + pubSubSourceSubscription);
+  }
+
+  @After(order = 1, value = "@PUBSUB_SOURCE_TEST")
+  public static void deleteSourcePubSubTopic() {
+    try {
+      PubSubClient.deleteTopic(pubSubSourceTopic);
+      BeforeActions.scenario.write("Deleted target PubSub topic " + pubSubSourceTopic);
+      pubSubSourceTopic = StringUtils.EMPTY;
+    } catch (Exception e) {
+      if (e.getMessage().contains("Invalid resource name given") || e.getMessage().contains("Resource not found")) {
+
+      }
+    }
+  }
+
+  @After(order = 2, value = "@PUBSUB_SCHEMA_TOPIC_TEST")
+  public static void deleteSourcePubSubSchemaTopic() {
+    try {
+      PubSubClient.deleteTopic(pubSubSourceTopic);
+      BeforeActions.scenario.write("Deleted target PubSub topic " + pubSubSourceTopic);
+      pubSubSourceTopic = StringUtils.EMPTY;
+    } catch (Exception e) {
+      if (e.getMessage().contains("Invalid resource name given") || e.getMessage().contains("Resource not found")) {
+
+      }
+    }
+  }
+
   @Before(order = 1, value = "@PUBSUB_SINK_TEST")
   public static void createTargetPubSubTopic() {
     pubSubTargetTopic = "cdf-e2e-test-" + UUID.randomUUID();
     BeforeActions.scenario.write("Target PubSub topic " + pubSubTargetTopic);
   }
 
+  @After(order = 1, value = "@PUBSUB_SCHEMA_TEST")
+  public static void deletePubSubSchema() throws IOException {
+    PubSubClient.deleteSchema(PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID), pubSubSchemaId);
```

Comment:
You are using ```PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID)``` multiple times in the code make a global variable for this and reuse it.

Reply:
directly used from pluginparameters file.